enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthScenariosLogic.java

```diff
@@ -404,10 +418,29 @@ public void roleManagement6() throws Throwable
 
         assertSuccess( henrik, "MATCH (n) RETURN n.name as name",
                 r -> {
-                    assertKeyIs( r, "name", "node0", "node1", "node2", "line1", "line2" );
+                    long numNodes = getObjectsAsList( r, "name" ).size();
+                    assertThat( numNodes, greaterThan( INITIAL_NUM_NODES ) );
+                    assertThat( numNodes, lessThan( INITIAL_NUM_NODES + ROWS_IN_CSV ) );
```

Comment:
I think that this scenario is over-specified. Currently we do not provide any guarantee of how much of a running periodic commit query that will be committed when you change the permissions - it could be all of it.